ssl/quic/quic_channel.c

```diff
@@ -2730,8 +2731,19 @@ int ossl_quic_channel_start(QUIC_CHANNEL *ch)
     ch->doing_proactive_ver_neg = 0; /* not currently supported */
 
     /* Handshake layer: start (e.g. send CH). */
-    if (!ossl_quic_tls_tick(ch->qtls))
+    if (!ossl_quic_tls_tick(ch->qtls)) {
+        uint64_t error_code;
+        const char *error_msg;
+        ERR_STATE *error_state = NULL;
+
+        if (ossl_quic_tls_get_error(ch->qtls, &error_code, &error_msg,
+                                    &error_state)) {
+            OSSL_ERR_STATE_restore(error_state);
```

Comment:
Can't we handle the same way as we do in ch_tick, i.e., raising a protocol error? Feels odd to do this in an ad-hoc way.

I also don't think we need to test the return value of tls_tick anymore, get_error suffices, again as in ch_tick.

Reply:
> Can't we handle the same way as we do in ch_tick, i.e., raising a protocol error? Feels odd to do this in an ad-hoc way.

I actually did have it that way originally but changed it to this, because I wasn't sure raising a protocol error was exactly right (this is an error that is occurring even before we've started sending protocol messages). But I'm ok with either way.

Reply:
Yeah. We already changed the state to ACTIVE above so there should be no issues using a protocol error here, so I'd recommend it.